File: lib-segment/src/main/java/websiteschema/mpsegment/dict/POS.java
package websiteschema.mpsegment.dict;

import java.io.Serializable;

public class POS implements Serializable, Comparable {

    public POS(String name, int count) {
        this.name = name;
        this.count = count;
    }

    @Override
    public int compareTo(Object obj) {
        if (obj != null && (obj instanceof POS)) {
            return name.compareTo(((POS) obj).name);
        } else {
            return 1;
        }
    }

    @Override
    public boolean equals(Object obj) {
        if (obj != null && (obj instanceof POS)) {
            return name.equals(((POS) obj).name);
        } else {
            return false;
        }
    }

    @Override
    public String toString() {
        StringBuilder stringbuffer = new StringBuilder();
        stringbuffer.append((new StringBuilder("\u8BCD\u6027\u540D\u79F0=")).append(name).append("\n").toString());
        stringbuffer.append((new StringBuilder("\u51FA\u73B0\u6B21\u6570=")).append(count).append("\n").toString());
        return stringbuffer.toString();
    }

    public String toDBFString() {
        StringBuilder stringbuffer = new StringBuilder();
        stringbuffer.append((new StringBuilder(" ")).append(name).toString());
        stringbuffer.append((new StringBuilder(" ")).append(count).toString());
        return stringbuffer.toString();
    }

    public int getCount() {
        return count;
    }

    public void setCount(int i) {
        count = i;
    }

    public String getName() {
        return name;
    }

    public void setName(String s) {
        name = s;
    }
    private String name;
    private int count;
}

File: lib-segment/src/main/scala/me/yingrui/segment/dict/POS.scala
package me.yingrui.segment.dict

class POS extends Comparable[POS] {

  var name = ""
  var count = 0

  override def compareTo(obj: POS): Int = {
    if (obj != null) {
      return name.compareTo(obj.name)
    } else {
      return 1
    }
  }

  def equals(obj: POS): Boolean = {
    if (obj != null) {
      return name.equals(obj.name)
    } else {
      return false
    }
  }

  override def toString(): String = {
    val stringBuilder = new StringBuilder()
    stringBuilder.append((new StringBuilder("\u8BCD\u6027\u540D\u79F0=")).append(name).append("\n").toString())
    stringBuilder.append((new StringBuilder("\u51FA\u73B0\u6B21\u6570=")).append(count).append("\n").toString())
    return stringBuilder.toString()
  }

  def toDBFString(): String = {
    val stringBuilder = new StringBuilder()
    stringBuilder.append((new StringBuilder(" ")).append(name).toString())
    stringBuilder.append((new StringBuilder(" ")).append(count).toString())
    return stringBuilder.toString()
  }

  def getCount(): Int = {
    return count
  }

  def setCount(i: Int) {
    count = i
  }

  def getName(): String = {
    return name
  }

  def setName(s: String) {
    name = s
  }
}

object POS {

  def apply(name: String, count: Int) = {
    val pos = new POS()
    pos.setCount(count)
    pos.setName(name)
    pos
  }
}